pkg/store/bucket.go

```diff
@@ -834,6 +825,62 @@ func (r *bucketIndexReader) preloadPostings() error {
 	return nil
 }
 
+// preloadPostings loads all given postings. It tries to avoid bigger gaps of irrelevant bytes.
+// TODO(bplotka): Merge this unit-tested function with the preloadFile (for chunks) and apply same for prealodSeries.
+func preloadPostings(
+	ctx context.Context,
+	postings []*lazyPostings,
+	loadPosting func(ctx context.Context, postings []*lazyPostings, start int64, length int64) error,
+) error {
+	if len(postings) == 0 {
+		return nil
+	}
+
+	sort.Slice(postings, func(i, j int) bool {
+		return postings[i].ptr.Start < postings[j].ptr.Start
+	})
+
+	// Maximum amount of irrelevant bytes between postings we are willing to fetch.
+	const maxPostingsGap = 1024
```

Comment:
This should be way higher. For some manual testing there's basically no difference between fetching 1 byte and 100KB. The per-request overhead outshines significantly here so probably at nearly 500KB of fetching overhead is at least what we want to allow.